src/libertem/udf/stddev.py

```diff
@@ -1,56 +1,183 @@
+    # Check if :cite:`Schubert2018` works in a future release
+    """
+    Given two sets of buffers, with sum of frames
+    and sum of variances, aggregate joint sum of frames
+    and sum of variances in destination buffers using one pass
+    algorithm.
+
+    This is ther numerical workhorse for :meth:`StdDevUDF.merge`.
+
+    Erich Schubert and Michael Gertz. Numerically stable parallel computation of
+    (co-)variance. In `Proceedings of the 30th International Conference on
+    Scientific and Statistical Database Management - SSDBM 18`. ACM Press, 2018.
+    `doi:10.1145/3221269.3223036 <https://doi.org/10.1145/3221269.3223036>`_
+
+    cite:Schubert2018
+
+    Parameters
+    ----------
+    dest_n : int
+        Number of frames aggregated in dest_sum and dest_varsum
+        The case :code:`dest_N == 0` is handled correctly.
+    dest_sum, dest_varsum : one-dimensional numpy.ndarray
+        Aggregation buffers, will be updated: sum of pixels, sum of variances
+    src_n : int
+        Number of frames aggregated in src_sum and src_varsum
+    src_sum, src_varsum : one-dimensional numpy.ndarray
+        Source buffers to merge: sum of pixels, sum of variances
+
+
+    Returns
+    -------
+    N:
+        New number of frames aggregated in aggregation buffer
+    """
+    if dest_n == 0:
+        dest_sum[:] = src_sum
+        dest_varsum[:] = src_varsum
+        return src_n
+    else:
+        n = dest_n + src_n
+
+        for pixel in range(len(dest_sum)):
+            dest_sum[pixel], dest_varsum[pixel] = merge_single(
+                n,
+                dest_n, dest_sum[pixel], dest_varsum[pixel],
+                src_n, src_sum[pixel], src_varsum[pixel], src_sum[pixel] / src_n
+            )
+    return n
+
+
+@numba.njit(fastmath=True)
+def process_tile(tile, n_0, sum_inout, varsum_inout):
+    '''
+    Compute sum and variance of :code:`tile` along navigation axis
+    and merge into aggregation buffers. Numerical "workhorse" for
+    :meth:`StdDevUDF.process_tile`.
+
+    Parameters
+    ----------
+    tile : 2-dimensional numpy.ndarray
+        Tile with flattened signal dimension
+    n_0 : int
+        Number of frames already aggegated in sum_inout, varsum_inout
+        Cannot be 0 -- the initial case is handled in :meth:`StdDevUDF.process_tile`
+        For unknown reasons, handling the case N0 == 0 in this function degraded performance
+        significantly. Re-check with a newer compiler version!
+    sum_inout, varsum_inout : numpy.ndarray
+        Aggregation buffers (1D) with length matching the flattened signal dimension
+        of the tile. The tile will be merged into these buffers (call by reference)
 
-    return VariancePart(var=var_AB, sum_im=sum_im_AB, N=N)
+    Returns
+    -------
+    n : int
+        New number of frames in aggregation buffer
+    '''
+    n_frames = tile.shape[0]
+    n_pixels = tile.shape[1]
+    n = n_0 + n_frames
+
+    BLOCKSIZE = 1024
```

Comment:
Re-testing for optimal block size with the latest iteration of the math kernel gave 256 as an universal optimum for float32, float64 and complex128 on my machine. :-) It seems that some other changes made it less sensitive to the size since 128, 512 gave good performance as well for all sizes.